src/main/test/CocktailControllerTest.java

```diff
@@ -0,0 +1,138 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Main.class)
+
+public class CocktailControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    CocktailRepository cocktailRepository;
+
+    @MockBean
+    private CocktailService cocktailService;
+
+
+    private Photo photoList = new Photo(1L, "http//google.com", "", "");
+    private List<Ingredient> ingredientList = Arrays.asList(new Ingredient(1L, "Wodka", "4 cl Wodka"), new Ingredient(2L, "Pfirsich Likör", "4 cl Pfirsich Likör"), new Ingredient(3L, "cranberry juice or Cranberry Rectal", "8 cl cranberry juice or Cranberry Rectal"), new Ingredient(4L, "orange juice", "8 cl of orange juice"), new Ingredient(5L, "ice", "ice cubes"));
+    private Cocktail cocktail = new Cocktail(1L, "Sex on the Beach Cocktail Recipe", "So I have to say one of the best drink's I've drunk so far has become my favorite drink as well", null, 1, 1, new User(1L, "Iurie Railean"), photoList, ingredientList);
+
+    private CocktailWebOutput cocktailWebOutput = new CocktailWebOutput(1l, COCKTAIL_NAME, "", null, new UserWeb(COCKTAIL_AUTHOR_NAME), Arrays.asList(new IngredientWeb(INGRDIENT_NAME, "some description")), 1, 1, null);
+    private List<CocktailWebOutput> cocktailWebOutputList = Arrays.asList(cocktailWebOutput);
+
+    private List<Cocktail> cocktailList = Arrays.asList(cocktail);
+
+    private final static String COCKTAIL_NAME = "Sex on the Beach Cocktail Recipe";
+    private final static String COCKTAIL_AUTHOR_NAME = "Iurie Railean";
+    private final static String INGRDIENT_NAME = "Wodka";
+
+    private final static String COCKTAIL_WEB_ARRAY_JSON = "[{\"id\":1,\"title\":\"Sex on the Beach Cocktail Recipe\",\"announce\":\"So I have to say one of the best drink's I've drunk so far has become my favorite drink as well\",\"publicationDate\":null,\"prepTime\":1,\"cookingTime\":1,\"author\":{\"id\":1,\"name\":\"Iurie Railean\"},\"image\":{\"id\":1,\"path\":\"http//google.com\",\"title\":\"\",\"alt\":\"\"},\"ingredients\":[{\"id\":1,\"title\":\"Wodka\",\"description\":\"4 cl Wodka\"},{\"id\":2,\"title\":\"Pfirsich Likör\",\"description\":\"4 cl Pfirsich Likör\"},{\"id\":3,\"title\":\"cranberry juice or Cranberry Rectal\",\"description\":\"8 cl cranberry juice or Cranberry Rectal\"},{\"id\":4,\"title\":\"orange juice\",\"description\":\"8 cl of orange juice\"},{\"id\":5,\"title\":\"ice\",\"description\":\"ice cubes\"}]}]";
```

Comment:
It is important to test against a json string you expect. If you test against a string, which is a result of several transformation, you cannot be sure that these transformations are correct and you can trust them, unless every mapper is tested and ObjectMapper configuration in your tests is exactly as in the production. Therefore, be explicit and compare an actual json string against a hardcoded expected json string, not against a json string as a result of some transformation.